### ISSUE TITLE
Default grid directory names

### ISSUE THREAD

**Issue by DanJohnsonCERN** [ganga-devs/ganga#1040]
Hi,

When Ganga submits to the GRID (in my case Dirac) it is very difficult to connect LFNs with Ganga jobs. This makes managing GRID space extremely difficult and time consuming for the user, and many don't bother, leading to enormous wastage of disk space. 

Please can we make a trivial change to include the Ganga job id (not even the subjob, necessarily) in the LFN path? It would be even better to have the name of the ganga job there too, but perhaps that is asking too much. 

It would be vastly preferable if this could happen by default.

Thanks!
Dan

**Comment by rob-c** [comment]
@DanJohnsonCERN I've added some code to #917 to store LFN using the following directory structure:

Eg: for a test job 4 from my repo:

OutputFiles are stored as:
```
/lhcb/user/r/rcurrie/GangaJob_4/OutputFiles/2017_07/171565/171565261/tmpYEqjgq.txt
```
InputFiles are uploaded as:
```
/lhcb/user/r/rcurrie/GangaJob_4/InputFiles/diracInputFiles_df64928a-dccd-455d-87cb-a3bdc69442b1.tgz
```

I've added this to store subjobs under the `GangaJob_4` folder which should make managing LFN from the base dir.

This should make sorting data by which Ganga job the file belongs to easier to parse as the Ganga job now has a folder under the users base lfn and all subjobs for large processings are collected together easier.

I think this should probably be included in 6.7.0 if everything goes well.

**Comment by rob-c** [comment]
Unfortunately we can't remove the extra folders `2017_07/171565/171565261` in the output path as this is required at some level by DIRAC but this solution should give some way of managing things better.

**Comment by DanJohnsonCERN** [comment]
Thanks very much, Rob, that's great, this is exactly what I was hoping for. Much appreciated.

**Comment by egede** [comment]
retest please

**Comment by DanJohnsonCERN** [comment]
I submitted a load of jobs yesterday early morning. When were the changes available?

On 18 Jul 2017 4:45 pm, Ulrik Egede <notifications@github.com> wrote:

retest please

**Comment by egede** [comment]
@DanJohnsonCERN The pull request #1072 is not merged yet. It needs to be merged and then go into a release. Also the default behaviour will stay as it currently us, so look in the release notes (when ready) for how to change it.

**Comment by DanJohnsonCERN** [comment]
Out of interest, why is the default not changed? It would make grid storage management much simpler-identification by short lived Dirac IDs is very difficult...
________________________________________

**Comment by mesmith75** [comment]
@DanJohnsonCERN There is quite a clear problem with this. If you have several ganga repositories (as I do when running toys) then the outputs are all going to pile into the same place with the same file names. Therefore I think having this not the default is wise - care does need to be taken.

**Comment by rob-c** [comment]
Quoting as for some reason the comment doesn't appear for me on github:
```
@DanJohnsonCERN There is quite a clear problem with this. If you have several ganga repositories (as I do when running toys) then the outputs are all going to pile into the same place with the same file names. Therefore I think having this not the default is wise - care does need to be taken.
  | Mark Smith
```
@mesmith75
I'm slightly concerned by this way of working tbh as you would up with conflicts from MassStorageFile automatic naming too (which imo is worse).
Is the motivation for this performance related or other?
Do you run multiple sets of toys at once in a given month? If not then the new naming scheme should be safe enough given the mandated month directory naming from DIRAC.

I think I can work in the ability to have `{jid}` and `{sjid}` into `DiracFile.remoteDir` (without too much code duplication) which would allow a user to do:

```
...
j.inputfile.remoteDir="toySet1_{jid}_{sjid}"
...
```

I think this would be the best of both worlds and fix #1055

**Comment by DanJohnsonCERN** [comment]
Hi Mark,

I too use multiple gangadirs but i disagree with you. The jobs will always be uniquely identified by Dirac Id - this just adds an extra directory layer and is an extra handle to figure out what work a job might be related to when it comes to clearing out time weeks (let's be realistic, sometimes months) later.

This proposal only adds information, even if some (vastly reduced) ambiguity remains.

Dan



On 19 Jul 2017 10:46 am, Mark Smith <notifications@github.com> wrote:

@DanJohnsonCERN<https://github.com/danjohnsoncern> There is quite a clear problem with this. If you have several ganga repositories (as I do when running toys) then the outputs are all going to pile into the same place with the same file names. Therefore I think having this not the default is wise - care does need to be taken.
